components/fxa-client/src/fxa_client.udl

```diff
@@ -0,0 +1,1075 @@
+
+
+
+// # Firefox Accounts Client
+//
+// The fxa-client component lets applications integrate with the
+// [Firefox Accounts](https://mozilla.github.io/ecosystem-platform/docs/features/firefox-accounts/fxa-overview)
+// identity service. The shape of a typical integration would look
+// something like:
+//
+// * Out-of-band, register your application with the Firefox Accounts service,
+//   providing an OAuth `redirect_uri` controlled by your application and
+//   obtaining an OAuth `client_id`.
+//
+// * On application startup, create a [`FirefoxAccount`] object to represent the
+//   signed-in state of the application.
+//     * On first startup, a new [`FirefoxAccount`] can be created by calling
+//       [`FirefoxAccount::new`] and passing the application's `client_id`.
+//     * For subsequent startups the object can be persisted using the
+//       [`to_json`](FirefoxAccount::to_json) method and re-created by
+//       calling [`FirefoxAccount::from_json`].
+//
+// * When the user wants to sign in to your application, direct them through
+//   a web-based OAuth flow using [`begin_oauth_flow`](FirefoxAccount::begin_oauth_flow)
+//   or [`begin_pairing_flow`](FirefoxAccount::begin_pairing_flow); when they return
+//   to your registered `redirect_uri`, pass the resulting authorization state back to
+//   [`complete_oauth_flow`](FirefoxAccount::complete_oauth_flow) to sign them in.
+//
+// * Display information about the signed-in user by using the data from
+//   [`get_profile`](FirefoxAccount::get_profile).
+//
+// * Access account-related services on behalf of the user by obtaining OAuth
+//   access tokens via [`get_access_token`](FirefoxAccount::get_access_token).
+//
+// * If the user opts to sign out of the application, calling [`disconnect`](FirefoxAccount::disconnect)
+//   and then discarding any persisted account data.
+
+namespace fxa_client {
+    
+};
+
+
+
+// Generic error type thrown by many [`FirefoxAccount`] operations.
+//
+// Precise details of the error are hidden from consumers, mostly due to limitations of
+// how we expose this API to other languages. The type of the error indicates how the
+// calling code should respond.
+//
+//
+//  `Authentication`:
```

Comment:
Any reason to not put the comments for each error next to the definition?

Reply:
TBH I don't recall, but the answer is almost certainly "I could not convince the automagical UDL generator to do that for me"...

Reply:
Aha, it was because we currently represent error enum variants in the `ComponentInterface` by just their string names, rather than as a `Variant` object, so there was nowhere for me to stash the docs on each individual variant. I'll fix it up by hand.